presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java

```diff
@@ -634,17 +634,52 @@ public void testInsertArray()
     }
 
     @Test
-    public void testDelete()
+    public void testUnsupportedDelete()
     {
-        // delete half the table, then delete the rest
+        if (supportsDelete()) {
+            return;
+        }
+
         String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM orders WITH NO DATA", 0);
+        assertQueryFails("DELETE FROM " + tableName, "This connector does not support updates or deletes");
+        assertUpdate("DROP TABLE " + tableName);
+    }
 
+    @Test
+    public void testDeleteEntireTable()
+    {
+        if (!supportsDelete()) {
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM orders", 15000);
+        assertUpdate("DELETE FROM " + tableName, 15000);
+        assertQuery("SELECT count(*) FROM " + tableName, "SELECT 0");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDeleteWithSimplePredicate()
+    {
         if (!supportsDelete()) {
-            assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM orders WITH NO DATA", 0);
-            assertQueryFails("DELETE FROM " + tableName, "This connector does not support updates or deletes");
-            assertUpdate("DROP TABLE " + tableName);
             return;
         }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM orders", 15000);
+        assertUpdate("DELETE FROM " + tableName + " WHERE orderkey = 1", 1);
```

Comment:
testDeleteWithSimplePredicate -> testDeleteWithBigintEquality

\+ add a test for varchar equality. Ideally one that would show incorrect query results when the condition is applied case insensitively.